File: 프로그래머스/unrated/181832. 정수를 나선형으로 배치하기/정수를 나선형으로 배치하기.java
class Solution {
    public int[][] solution(int n) {
        int[][] answer = new int[n][n];
        
        int x = 1;
        int cols = 0;
        int rows = 0;
        int cole = n-1;
        int rowe = n-1;
            
        while (x <= n*n) {
            for (int i=cols; i<=cole; i++) {
                answer[rows][i] = x++;
            }
            
            rows++;
            
            for (int i=rows; i<=rowe; i++) {
                answer[i][cole] = x++;
            }
            
            cole--;
            
            for (int i=cole; cols<=i; i--) {
                answer[rowe][i] = x++;
            }
            
            rowe--;
            
            for (int i=rowe; rows<=i; i-- ) {
                answer[i][cols] = x++;
            }
            
            cols++;
        }
        
        return answer;
    }
}

File: 프로그래머스/unrated/181832. 정수를 나선형으로 배치하기/정수를 나선형으로 배치하기.py
def solution(n):
    answer = [[0]*n for i in range(n)] #[[None for j in range(n)] for i in range(n)] #[[0]*n]*n ?
    
    move = [[0,1], [1,0], [0,-1], [-1,0]]
    x, y, m = 0, 0, 0
    
    for i in range(1, n*n+1):
        answer[y][x] = i
        
        if y + move[m][0] >= n or x + move[m][1] >= n or answer[y + move[m][0]][x + move[m][1]]:
            m = (m+1)% len(move)
        
        y = y + move[m][0]
        x = x + move[m][1]
    
    return answer